x-pack/platform/packages/shared/onechat/onechat-common/base/errors.ts

```diff
@@ -164,6 +165,25 @@ export const createConversationNotFoundError = ({
   );
 };
 
+/**
+ * Represents an internal error
+ */
+export type OnechatRequestAbortedError = OnechatError<OnechatErrorCode.requestAborted>;
+
+/**
+ * Checks if the given error is a {@link OnechatRequestAbortedError}
+ */
+export const isRequestAbortedError = (err: unknown): err is OnechatRequestAbortedError => {
+  return isOnechatError(err) && err.code === OnechatErrorCode.requestAborted;
+};
+
+export const createRequestAbortedError = (
+  message: string,
+  meta?: Record<string, any>
+): OnechatRequestAbortedError => {
+  return new OnechatError(OnechatErrorCode.requestAborted, message, meta ?? {});
```

Comment:
not part of this PR: but should we add OnechatError `toString` (or whatever we have in js) method? To include the message? 

E.g. https://github.com/elastic/kibana/blob/main/x-pack/platform/plugins/shared/onechat/public/application/components/agents/edit/agent_form.tsx#L67 upon validation error this would only print `BadRequestError` without the message with validation error ... I actually had to go to devconsole to inspect that request and read the message that way.